pkg/istio-agent/sds-agent.go

```diff
@@ -534,66 +428,28 @@ func (sa *Agent) newSecretCache(serverOptions sds.Options) (workloadSecretCache
 	fetcher.UseCaClient = true
 	fetcher.CaClient = caClient
 
-	workloadSdsCacheOptions.TrustDomain = serverOptions.TrustDomain
-	workloadSdsCacheOptions.Pkcs8Keys = serverOptions.Pkcs8Keys
-	workloadSdsCacheOptions.OutputKeyCertToDir = serverOptions.OutputKeyCertToDir
-
 	return
 }
 
 // TODO: use existing 'sidecar/router' config to enable loading Secrets
-func newGatewaySecretCache(namespace string) (gatewaySecretCache *cache.SecretCache) {
+func (sa *Agent) newSecretCache(namespace string) (gatewaySecretCache *cache.SecretCache) {
 	gSecretFetcher := &secretfetcher.SecretFetcher{
 		UseCaClient: false,
 	}
+	// TODO: use the common init !
 	// If gateway is using file mounted certs, we do not have to setup secret fetcher.
-	if shouldProvisionCertificates() {
-		cs, err := kube.CreateClientset("", "")
-
-		if err != nil {
-			log.Errorf("failed to create secretFetcher for gateway proxy: %v", err)
-			os.Exit(1)
-		}
-		gSecretFetcher.FallbackSecretName = "gateway-fallback"
-
-		gSecretFetcher.InitWithKubeClientAndNs(cs.CoreV1(), namespace)
-
-		gatewaySecretChan = make(chan struct{})
-		gSecretFetcher.Run(gatewaySecretChan)
+	cs, err := kube.CreateClientset("", "")
+	if err != nil {
+		log.Errorf("failed to create secretFetcher for gateway proxy: %v", err)
+		os.Exit(1)
 	}
-	gatewaySecretCache = cache.NewSecretCache(gSecretFetcher, sds.NotifyProxy, gatewaySdsCacheOptions)
-	return gatewaySecretCache
-}
-
-func applyEnvVars() {
-	serverOptions.PluginNames = strings.Split(pluginNamesEnv, ",")
 
-	serverOptions.EnableWorkloadSDS = true
+	gSecretFetcher.FallbackSecretName = "gateway-fallback"
 
-	serverOptions.EnableGatewaySDS = enableGatewaySDSEnv
+	gSecretFetcher.InitWithKubeClientAndNs(cs.CoreV1(), namespace)
 
-	serverOptions.CAProviderName = caProviderEnv
-
-	// TODO: extract from ProxyConfig
-	serverOptions.TrustDomain = trustDomainEnv
-	serverOptions.Pkcs8Keys = pkcs8KeysEnv
-	serverOptions.ECCSigAlg = eccSigAlgEnv
-	serverOptions.RecycleInterval = staledConnectionRecycleIntervalEnv
-	workloadSdsCacheOptions.ECCSigAlg = eccSigAlgEnv
-	workloadSdsCacheOptions.SecretTTL = secretTTLEnv
-	workloadSdsCacheOptions.SecretRotationGracePeriodRatio = secretRotationGracePeriodRatioEnv
-	workloadSdsCacheOptions.RotationInterval = secretRotationIntervalEnv
-	workloadSdsCacheOptions.InitialBackoffInMilliSec = int64(initialBackoffInMilliSecEnv)
-	// Disable the secret eviction for istio agent.
-	workloadSdsCacheOptions.EvictionDuration = 0
-	if citadel.ProvCert != "" {
-		workloadSdsCacheOptions.AlwaysValidTokenFlag = true
-	}
-	workloadSdsCacheOptions.OutputKeyCertToDir = serverOptions.OutputKeyCertToDir
-}
-
-// shouldProvisionCertificates returns true if certs needs to be provisioned for the workload/gateway.
-// Returns flase, when proxy uses file mounted certificates i.e. FILE_MOUNTED_CERTS is true.
-func shouldProvisionCertificates() bool {
```

Comment:
Is this still covered? I think we had tests for it so maybe so if those pass...

Reply:
I moved the check inline - it was effectively just the env variable check, and I don't think we use it anywhere now. It's for the case /etc/certs are mounted from a different CA. The method name was pretty bad - has nothing to do with the fact that we use 'externally  managed certificate'. Also added more docs on the field in the merged Options.